File: 01_two_sums/java/Solution.java
import java.util.HashMap;

public class Solution {

    public int[] twoSum(int[] nums, int target) {
        final var map = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; ++i) {
            int complement = target - nums[i];
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }
            map.put(nums[i], i);
        }
        return new int[] { -1, -1 };
    }
}

File: 01_two_sums/python/solution.py
#!/usr/bin/env python3
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for hop in range(1, len(nums)):
            for i, first in enumerate(nums):
                if i + hop >= len(nums):
                    break
                second = nums[i + hop]
                if (first + second) == target:
                    return [i, i + hop]
        return Nil

